### PR TITLE
Make association assets names more consistent

### PR DESCRIPTION
**Pull request by bendnorman**
# PR Overview
This PR makes association assets more consistent by alphabetizing the linked sources and making sure linked sources come before linked entity names. See #3030 for more details.

# PR Checklist

- [ ] Merge the most recent version of the branch you are merging into (probably `dev`).
- [ ] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [ ] Make sure you've included good docstrings.
- [ ] For major data coverage & analysis changes, [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)
- [ ] Include unit tests for new functions and classes.
- [ ] Defensive data quality/sanity checks in analyses & data processing functions.
- [ ] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html) and reference reference the PR and related issues.
- [ ] Do your own explanatory review of the PR to help the reviewer understand what's going on and identify issues preemptively.

### CODE REVIEW

File: src/pudl/transform/eia.py [hunk 1]
```diff
@@ -1217,15 +1217,18 @@ def harvested_entity(context, **clean_dfs):
...
+    table_name: str, _core_table_name: str, io_manager_key: str | None = None
 ) -> AssetsDefinition:
-    """An asset factory for finished EIA tables."""
-    # TODO (bendnorman): Create a more graceful function for parsing table name
-    table_name_parts = table_name.split("__")
-    dataset = table_name_parts[0].replace("core_", "")
-    table_name_no_asset_type = "_".join(table_name_parts[-1].split("_")[1:])
+    """An asset factory for finished EIA tables.
 
-    _core_table_name = f"_core_{dataset}__{table_name_no_asset_type}"
```
Comment (bendnorman) [inline src/pudl/transform/eia.py:1228]: I figured creating a mapping of pre-harvested assets to harvested assets was better than my janky asset naming parsing.

**Comment by bendnorman** [comment]
I decided to just recreate all of the migrations again because we're I ran into some migration conflicts and we're going to recreate them before we merge `rename-core-asset` into dev anyways.

**Review by zaneselvans — CHANGES_REQUESTED** [review]
Only minor internal inconsistency I noticed was that now the `epacamd_eia.py` module is named differently from the reordered assets that it helps produce, and it seemed like having them named the same would be more intuitive.

**Comment by bendnorman** [comment]
Looks like there are a lot of instances of `epacems_eia` being used for variables, module names, and the archive. How difficult would it be to rename the archive? I think it makes more sense to rename just the assets in this PR and do a deeper replacement of `epacems_eia` with `eia_epacems` in a separate PR. We're planning on doing this with [references to mcoe](https://github.com/catalyst-cooperative/pudl/issues/2910).

**Review by zaneselvans — APPROVED** [review]

**Pull request merged**